public/app/features/inspector/InspectDataTab.tsx

```diff
@@ -227,7 +223,7 @@ export class InspectDataTab extends PureComponent<Props, State> {
                 >
                   <Switch
                     value={!!options.withTransforms}
-                    onChange={() => onOptionsChange({ ...options, withTransforms: !options.withTransforms })}
+                    onChange={() => onOptionsChange!({ ...options, withTransforms: !options.withTransforms })}
```

Comment:
Not sure why, but event though we check if **showPanelTransformationsOption** is truthy - `{showPanelTransformationsOption && (` and in **showPanelTransformationsOption** we are checking that we have **onOptionsChange** `const showPanelTransformationsOption = onOptionsChange &&`, typescript was still throwing warning.

Reply:
I don't think Typescript knows if onOptionsChange is defined here. Previously the code returned null when onOptionsChange wasn't defined.

Reply:
Not sure this is part of the scope for this PR but I think the `renderDataOptions` deserves its own React component(s). The `InspectDataTab` class is just getting bigger if we don't refactor it soon.

Reply:
Yeah, not scope of this PR, but I totally agree.